web/src/search_suggestion.ts

```diff
@@ -709,82 +785,83 @@ function get_operator_suggestions(last: NarrowTerm): Suggestion[] {
 }
 
 class Attacher {
-    result: Suggestion[] = [];
+    result: SuggestionPart[] = [];
     prev = new Set<string>();
-    base: Suggestion;
+    base: SuggestionPart;
 
-    constructor(base: Suggestion) {
+    constructor(base: SuggestionPart) {
         this.base = base;
     }
 
-    prepend_base(suggestion: Suggestion): void {
-        if (this.base && this.base.description_html.length > 0) {
-            suggestion.search_string = this.base.search_string + " " + suggestion.search_string;
-            suggestion.description_html =
-                this.base.description_html + ", " + suggestion.description_html;
+    prepend_base_and_restructure_suggestion(
+        suggestion: SuggestionPart,
+        prepend_base?: boolean,
+    ): void {
+        if (prepend_base && this.base) {
+            suggestion.suggestion_parts = [
+                ...this.base.suggestion_parts,
+                ...suggestion.suggestion_parts,
+            ];
+            if (this.base.search_string && suggestion.search_string) {
+                suggestion.search_string = this.base.search_string + " " + suggestion.search_string;
+            }
         }
     }
 
-    push(suggestion: Suggestion): void {
+    push(suggestion: SuggestionPart, prepend_base?: boolean): void {
+        this.prepend_base_and_restructure_suggestion(suggestion, prepend_base);
         if (!this.prev.has(suggestion.search_string)) {
             this.prev.add(suggestion.search_string);
             this.result.push(suggestion);
         }
     }
 
-    push_many(suggestions: Suggestion[]): void {
+    push_many(suggestions: SuggestionPart[], prepend_base?: boolean): void {
         for (const suggestion of suggestions) {
-            this.push(suggestion);
-        }
-    }
-
-    attach_many(suggestions: Suggestion[]): void {
-        for (const suggestion of suggestions) {
-            this.prepend_base(suggestion);
-            this.push(suggestion);
+            this.push(suggestion, prepend_base);
         }
     }
 }
 
-export function get_search_result(query: string): Suggestion[] {
-    let suggestion: Suggestion;
+export function get_search_result(query: string): SuggestionPart[] {
+    let suggestion: SuggestionPart;
 
-    // search_terms correspond to the terms for the query in the input.
+    // search_operators correspond to the operators for the query in the input.
     // This includes the entire query entered in the searchbox.
-    // terms correspond to the terms for the entire query entered in the searchbox.
-    const search_terms = Filter.parse(query);
+    // operators correspond to the operators for the entire query entered in the searchbox.
+    const search_operators = Filter.parse(query);
```

Comment:
So this bit of change looks to be just renaming `search_terms` to `search_operators`; can you do that replacement as a separate prep commit? It really improves readability if mechanical changes like that can be separated from the main thing happening.

Reply:
I wasn't sure if `SuggestionPart` was within the scope of `_terms`, but I have omitted it and added it to `_terms`.